### PR TITLE
Add stub test using the shared CI node

### PR DESCRIPTION
**Pull request by shlevy**

### CODE REVIEW

**Comment by ladamesny** [comment]
Hi @shlevy , thanks for this. Can I make the change directly on this PR?

Before running the test command, I need to set three variables:

1) CARDANO_NODE_SOCKET_PATH
2) CARDANO_TESTNET_MAGIC
3) TREASURY

I see the socket path is getting set, so I need help on the other two. 

TREASURY should be set to the "directory which should be used as a store for test suite treasury account. We expect two files there: \`payment.skey\`, \`payment.vkey\`". I have some key files that we can use, but can you help me store them securely and figure out what path I should set here?

Thanks again!

**Review by ladamesny — APPROVED** [review]

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -103,3 +103,51 @@ jobs:
           # Disabled until we can find a way to run it only for Linux
           # nix-shell --command "updateMaterialized" --option trusted-public-keys "${{ env.TRUSTED_PUBLIC_KEYS }}" --option substituters "${{ env.SUBSTITUTERS }}"
           git diff --exit-code
+  shared-node-stub:
```
Comment (shlevy) [inline .github/workflows/test.yml:106]: This should be renamed

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -80,3 +80,50 @@ jobs:
...
+          cat > keys/payment.addr <<EOF
+          $TREASURY_ADDR
+          EOF
+          cat > keys/payment.vkey <<EOF
+          $TREASURY_VKEY
+          EOF
+          cat > keys/payment.skey <<EOF
+          $TREASURY_SKEY
+          EOF
+          nix-shell -p socat --command "socat UNIX-LISTEN:node.sock,fork OPENSSL-CONNECT:dev-ci-node.marlowe-finance.io:443,cert=client.pem,cafile=server.crt,openssl-commonname=marlowe-node-server,snihost=dev-ci-node.marlowe-finance.io" &>socat-log &
```
Comment (ladamesny) [inline .github/workflows/test.yml:125]: hi @shlevy I just updated this branch with main this morning. Do we need to update any of this now that we are using nix flakes?

If we do need to update for nix flakes, will that update install marlowe-cli on the ci for us?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -38,3 +38,50 @@ jobs:
...
+          EOF
+          cat > keys/payment.vkey <<EOF
+          $TREASURY_VKEY
+          EOF
+          cat > keys/payment.skey <<EOF
+          $TREASURY_SKEY
+          EOF
+          nix develop -p socat --command "socat UNIX-LISTEN:node.sock,fork OPENSSL-CONNECT:dev-ci-node.marlowe-finance.io:443,cert=client.pem,cafile=server.crt,openssl-commonname=marlowe-node-server,snihost=dev-ci-node.marlowe-finance.io" &>socat-log &
+          export CARDANO_NODE_SOCKET_PATH=node.sock
+          export CARDANO_TESTNET_MAGIC=1566
```
Comment (ladamesny) [inline .github/workflows/test.yml:85]: change to preview which might be 2 magic.

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -38,3 +38,50 @@ jobs:
...
+          export TREASURY=keys/
+          nix develop --command "cabal install exe:marlowe-cli && cd marlowe-cli && . ./run-nonpab-tests.sh"
```
Comment (ladamesny) [inline .github/workflows/test.yml:87]: Might not need to install cabal part of this line as nix develop already install the marlowe cli for us.

**Comment by jhbertra** [comment]
Can this be closed @shlevy @ladamesny - it is pretty old and I assume that it no longer aligns with current testing plans.

**Pull request closed without merging**